api/v3/Membership.php

```diff
@@ -42,7 +59,17 @@
  *   API result array.
  */
 function civicrm_api3_membership_delete($params) {
-  return _civicrm_api3_basic_delete(_civicrm_api3_get_BAO(__FUNCTION__), $params);
+  if (isset($params['preserve_contribution'])) {
+    if (CRM_Member_BAO_Membership::del($params['id'], $params['preserve_contribution'])) {
+      return civicrm_api3_create_success(TRUE, $params);
```

Comment:
hi @joseltorres looking over this again, I think what might be the best idea here is to actually remove the isset and just have _civicrm_api3_basic_delete(_civicrm_api3_get_BAO(__FUNCTION__), $params); Then in the CRM_Membership_BAO_Membership::del() function have $preserveContrib = isset($params['preserve_contribution']) ? $params['preserve_contribution'] : FALSE; 

That way then we aren't doing anything strange at the API level and getting it all to be handled at the BAO level